### DISCUSSION TITLE
How are real time events handled?

### DISCUSSION THREAD

**Discussion by rlippmann** [homebridge-simplisafe3/homebridge-simplisafe3#438] [Q&A]
Hi,

I'm actually a developer who does home assistant stuff, and that integration polls every 60 seconds.  Which is kind of useless for automations.

The HA integration does connect to a websocket interface (which I guess I need a developer license or something to obtain).

Not being a JS developer, I was wondering how you manage to get real time events?  It looks like you're polling a REST API really quickly, but I'm not sure.

TIA,
Rob

**Comment by shamoon** [comment]
This uses web sockets, the entire point of them is to have 'real-time' messaging. We supplement with polling for some things

**Comment by rlippmann** [comment]
So the web sockets will push real time sensor updates?  Both open and close for things like door sensors?

How does one get access to the websocket API, or was it reverse engineered?

**Reply by shamoon** [reply]
Yes x2

This is open source, after all...

see https://github.com/homebridge-simplisafe3/homebridge-simplisafe3/blob/86b2c119e8209601ac418483406660ae56722d65/src/simplisafe.js#L394

**Comment by rlippmann** [comment]
So, going through the code, an entry door open/close isn't done in real time.  Correct?

**Reply by shamoon** [reply]
Yes I believe thats true there is no websocket event for entry sensor open / close

**Reply by rlippmann** [reply]
I was thinking of using secret alerts to implement that.  Would send the alerts to a throwaway email address.  The main email will actually get the fact that a secret alert was triggered.

Won't do anything for sensor close, but it's better than waiting 60 seconds for the poll.

**Comment by rlippmann** [comment]
It looks like you're also using secret alerts for motion detectors.  Are they only set for those, or is it also for entry door secret alerts?

**Reply by shamoon** [reply]
I think so, you're asking about stuff I wrote 4 years ago...

**Reply by rlippmann** [reply]
> I think so, you're asking about stuff I wrote 4 years ago...

Wow, they haven't changed in 4 years? lol
